### ISSUE TITLE
Gnome Tracker: no miners

### ISSUE THREAD

**Issue by bug-buster-yom** [NixOS/nixpkgs#53110]
## Issue description

Installed NixOS 18.09 on several laptops, with Gnome. All Tracker dependant functions won't work (e.g. nothing to show in both Gnome Documents and Gnome Music).

Tracker is installed, but there are no miners available:

`tracker daemon --list-miners-available`

`0 collecteur est installé`

System installed in French, the message means that 0 miners are installed. Still, there IS a message from Tracker, which means that Tracker itself is installed.

### Steps to reproduce

Install NixOS 18.09 with:

- `services.xserver.displayManager.gnome3.enable = true;`
- `services.gnome3.tracker.enable = true;`
- `services.gnome3.tracker-miners.enable = true;`

## Technical details

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the
results.

- `system: "x86_64-linux"`
- `host os: Linux 4.14.88, NixOS, 18.09.1676.7e88992a8c7 (Jellyfish)`
- `multi-user?: yes`
- `sandbox: yes`
- `version: nix-env (Nix) 2.1.3`
- `nixpkgs: /nix/var/nix/profiles/per-user/root/channels/nixos`

**Comment by hedning** [comment]
I took look (though on unstable with gnome 3.30).

`tracker daemon --list-miners-available` lists zero miners here as well, but the services are running and eg. gnome-photos and gnome-music is working. So I suspect this isn't actually a problem.

It's worth double checking that you've turned on tracking for the relevant directories, either through `Settings` -> `Search` -> gear at the bottom right. Or doing something like this `dconf write /org/freedesktop/tracker/miner/files/index-recursive-directories ['&PICTURES', '&DOCUMENTS', '&MUSIC']`.

Now, `gnome-documents` isn't working here either. But it looks like this is the problem
> des. 31 10:15:28 x1 .gnome-document[18107]: Failed to load shared library '/nix/store/ph28v47rmjs6mvnddxmskj1cgakjf67l-gnome-documents-3.30.0/lib/gnome-documents/libgdprivate-1.0.so' referenced by the typelib: libgd.so: cannot open shared object file: No such file or directory

This seems to result from broken linking:
```
cd $(nix eval nixpkgs.gnome3.gnome-documents --raw)/lib/gnome-documents
❯ ldd libgdprivate-1.0.so | rg libgd
	libgd.so => not found
        ...
```

@jtojnar have filed an upstream issue for this: https://gitlab.gnome.org/GNOME/gnome-documents/issues/5 We should probably apply his mentioned fix.

**Comment by jtojnar** [comment]
IIRC, the miners are picked up from `XDG_DATA_DIRS`, and we cannot add them to tracker because that would cause a dependency cycle. Maybe we could add the miners to pathsToLink in the tracker module.

**Comment by bug-buster-yom** [comment]
> It's worth double checking that you've turned on tracking for the relevant directories, either through Settings -> Search -> gear at the bottom right.

All switches are turned on. The gear at the bottom right is here, but disabled.

> Or doing something like this `dconf write /org/freedesktop/tracker/miner/files/index-recursive-directories ['&PICTURES', '&DOCUMENTS', '&MUSIC']`

~~~
error: 1:expected value

Usage:
  dconf write KEY VALUE 

Write a new value to a key

Arguments:
  KEY         A key path (starting, but not ending with '/')
  VALUE       The value to write (in GVariant format)
~~~

**Comment by austinbutler** [comment]
This is still confusing. `tracker daemon --list-miners-available` says none available and Settings -> Search -> Search Locations is greyed out. But there's a tracker process running.

```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.9.1, NixOS, 21.03pre248170.007126eef72 (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.7`
 - channels(root): `"nixos-21.03pre248170.007126eef72, home-manager"`
 - channels(austin): `"home-manager"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by benneti** [comment]
I also have this problem but I only have a tracker3 command (gnome-music still shows my music) and I cannot change the search locations in gnome settings either.
(Maybe relevant is that I changed my XDG_DIRECTORIES using home-manager which creates `.config/user-dirs.conf`with `enabled=False` to not overwrite my custom directories on every login)

 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.13, NixOS, 21.03.20210208.8c87313 (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.4pre20201205_a5d85d0`
 - nixpkgs: `/nix/store/milbcj3ck45mipsi0xmgx29a42qx6h09-source`

**Comment by jtojnar** [comment]
It works for me now:

```ShellSession
$ result/libexec/tracker3/daemon --list-miners-available
Found 1 miner installed:
  org.freedesktop.Tracker3.Miner.Files
```

The command is not available by default, tracked by #153378.
